test/e2e/framework.go

```diff
@@ -938,6 +943,58 @@ func (data *TestData) deployFlowAggregator(ipfixCollector string, o flowVisibili
 	if err = data.podWaitForReady(2*defaultTimeout, flowAggPod.Name, flowAggregatorNamespace); err != nil {
 		return err
 	}
+	// Check recordmetrics of Flow Aggregator to make sure Antrea-agent Pods/ClickHouse/IPFIX collector and Flow Aggregator
+	// are correctly connected
+	if err := data.getAndCheckFlowAggregatorMetrics(flowAggPod.Name); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (data *TestData) getAndCheckFlowAggregatorMetrics(podName string) error {
+	errLogs := []string{}
+	if err := wait.Poll(defaultInterval, defaultTimeout, func() (bool, error) {
+		localConfig := restclient.CopyConfig(data.kubeConfig)
+		listenAddress := "localhost"
+		listenPort := apis.FlowAggregatorAPIPort
+		pf, err := portforwarder.NewPortForwarder(localConfig, flowAggregatorNamespace, podName, apis.FlowAggregatorAPIPort, "localhost", listenPort)
+		if err != nil {
+			errLogs = append(errLogs, fmt.Sprintf("error when port-forwarding Flow Aggregator Pod: %s/%s, err: %v", flowAggregatorNamespace, podName, err))
+			return false, nil
+		}
+		pf.Start()
+		defer pf.Stop()
+		localConfig.Host = net.JoinHostPort(listenAddress, fmt.Sprint(listenPort))
+		localConfig.Insecure = true
+		localConfig.CAFile = ""
+		localConfig.CAData = nil
+		clientset, err := kubernetes.NewForConfig(localConfig)
+		if err != nil {
+			errLogs = append(errLogs, fmt.Sprintf("error when creating Flow Aggregator client: %v", err))
+			return false, nil
+		}
+		jsonMetrics, err := clientset.RESTClient().
```

Comment:
can you remind me why we don't use `antctl` (from the FA Pod) here?

Reply:
And I think we prefer calling k8s API instead of running RunCommandFromPod, which will be used for running antclt?

https://github.com/antrea-io/antrea/pull/5770#discussion_r1433079962

Reply:
K8s API is definitely preferred (and in particular preferred to using `kubectl` with `RunCommandOnNode`). But it's not worth it if you have to set up port forwarding. You are not really calling K8s API in this case. `RunCommandFromPod` with `antctl` is fine and portable, and we use it in several places. It should be much less verbose.